src/ui/Search/searchinfilesworker.cpp

```diff
@@ -89,40 +91,18 @@ void SearchInFilesWorker::run()
             }
         } while (retry);
 
-        QString content = decodedText.text;
-
-        // Search result structure
         FileSearchResult::FileResult structFileResult;
-        structFileResult.fileName = fileName;
 
         // Run the search
-        QRegularExpressionMatchIterator i = regex.globalMatch(content);
-        while (i.hasNext())
-        {
-            //emit progress(fileName + "\n" + QString::number(curFileMatches));
-            m_stopMutex.lock();
-            bool stop = m_stop;
-            m_stopMutex.unlock();
-            if (stop) {
-                f.close();
-                emit finished(true);
-                return;
-            }
-
-            QRegularExpressionMatch match = i.next();
-            QStringList matches = match.capturedTexts();
-
-            if (!matches[0].isEmpty()) {
-                structFileResult.results.append(buildResult(match, &content));
-
-                curFileMatches++;
-                //totalFileMatches++;
-            }
+        if (m_regex.pattern().contains(QStringLiteral("\\n"))) {
```

Comment:
The result of the `contains` call could be computed once before the loop instead of being computed once every iteration.